lib/_http_outgoing.js

```diff
@@ -122,43 +125,49 @@ OutgoingMessage.prototype._send = function(data, encoding, callback) {
 
 
 OutgoingMessage.prototype._writeRaw = function(data, encoding, callback) {
-  if (util.isFunction(encoding)) {
+  if (typeof encoding === 'function') {
     callback = encoding;
     encoding = null;
   }
 
   if (data.length === 0) {
-    if (util.isFunction(callback))
+    if (typeof callback === 'function')
       process.nextTick(callback);
     return true;
   }
 
-  if (this.connection &&
-      this.connection._httpMessage === this &&
-      this.connection.writable &&
-      !this.connection.destroyed) {
+  var connection = this.connection;
+  var output, outputLength, outputEncodings, outputCallbacks;
```

Comment:
Hoisting variable declarations outside the blocks where they are assigned/used is somewhat unidiomatic.  Just a minor nit, though.